arm/Microsoft.Network/azureFirewalls/deploy.bicep

```diff
@@ -102,18 +108,6 @@ param diagnosticMetricsToEnable array = [
   'AllMetrics'
 ]
 
-var ipConfigurations_var = [for ipConfiguration in ipConfigurations: {
```

Comment:
Can there only be one IP configuration per firewall?

It used to be an array (and the property still is) but we only have one entry.

Reply:
I assume this comes down to the creation of the public IP that would otherwise have been provided in the array?

Reply:
The template shows there can be many ip configurations per firewall according to the ipconfigurations object which can take an array like this
<img width="146" alt="image" src="https://user-images.githubusercontent.com/19974635/165531222-161d8ce4-1237-44ae-9a77-8c4388d8b702.png">

https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2021-05-01/azurefirewalls?tabs=bicep#template-format

so can the bastionhosts which is where I copied the logic from
https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2021-05-01/bastionhosts?tabs=bicep#template-format

When I look at the portal, you can add many public ips
<img width="815" alt="image" src="https://user-images.githubusercontent.com/19974635/165532584-3a77a17f-9651-4df7-9a04-55527fc12419.png">

However, there's not a portal view for assigning multiple subnets to an azurefirewall
<img width="251" alt="image" src="https://user-images.githubusercontent.com/19974635/165532824-39c612ac-2cc7-41cf-8b83-25a50c7afc96.png">
I can test this locally if it'll complain.

Reply:
After I test, we can see if it's worth merging ipconfigurations with the default public ip created in the module

Reply:
Update: It's not possible to have more than one subnet reference
![image](https://user-images.githubusercontent.com/19974635/166317595-fa83b474-5530-437a-87f9-3e8ef20a3444.png)

You can have multiple public ips, but I need to check if the CAF documentation is against maintaining multiple public ips for one resource

Reply:
The general microsoft documentation say's there's use cases for DNAT and SNAT here
https://docs.microsoft.com/en-us/azure/firewall/deploy-multi-public-ip-powershell#:~:text=Azure%20Firewall%20with%20multiple%20public%20IP%20addresses%20is,and%20remove%20public%20IP%20addresses%20for%20Azure%20Firewall.

Reply:
Note: you MUST provide a public ip resource id when providing a subnet Resource Id based on this test

```
![image](https://user-images.githubusercontent.com/19974635/166552796-40f3e649-5394-428b-926d-4af6b8c6fea1.png)

```

So if a subnet id address is provided without a public ip, then the public ip address should be applied to that subnet id. Also, there cannot be multiple subnet ids as shown in the previous example - so I need to create a parser that finds the first subnet id in ipconfigurations and merges the public ip address to create to that object

Reply:
Also you cannot deploy without a subnet in ipconfigurations
<img width="716" alt="image" src="https://user-images.githubusercontent.com/19974635/168082419-d1ed5de0-284c-4fc6-ac02-9084652658b2.png">

Reply:
and you're required to have at least one pip
<img width="688" alt="image" src="https://user-images.githubusercontent.com/19974635/168090766-4ece036e-f446-4cc6-aab1-3b3fa5ec6f68.png">

Reply:
<img width="745" alt="image" src="https://user-images.githubusercontent.com/19974635/168151031-a3ae2acd-61d2-4293-b89c-675a23700872.png">


This is what happens when you do 
```
          publicIPAddress: {
            //Use existing public ip, new public ip created as in this module, or none if isCreateDefaultPublicIP is false
            id: !(empty(azureFirewallSubnetPublicIpId)) ? azureFirewallSubnetPublicIpId : (isCreateDefaultPublicIP ? publicIPAddress.outputs.resourceId : null)
          }

```

But it works when you completely take out the publicIpAddress object

So, I need to find out a way to build the ipconfigurations with publicIpAddress and without dependeing on isCreateDefaultPublicIp